src/fmu/dataio/_objectdata_provider.py

```diff
@@ -118,43 +118,10 @@ def npfloat_to_float(v: Any) -> Any:
 
 @dataclass
 class DerivedObjectDescriptor:
-    subtype: Literal[
-        "RegularSurface",
-        "Polygons",
-        "Points",
-        "RegularCube",
-        "CPGrid",
-        "CPGridProperty",
-        "DataFrame",
-        "JSON",
-        "ArrowTable",
-    ]
-    classname: Literal[
-        "surface",
-        "polygons",
-        "points",
-        "cube",
-        "cpgrid",
-        "cpgrid_property",
-        "table",
-        "dictionary",
-    ]
-    layout: Literal[
-        "regular",
-        "unset",
-        "cornerpoint",
-        "table",
-        "dictionary",
-    ]
-    efolder: Literal[
-        "maps",
-        "polygons",
-        "points",
-        "cubes",
-        "grids",
-        "tables",
-        "dictionaries",
-    ]
+    subtype: types.Subtype
```

Comment:
This should have been part of https://github.com/equinor/fmu-dataio/pull/484 but i slipped thru during the rebase/cherry-picking phase.